flink-cdc-runtime/src/main/java/com/ververica/cdc/runtime/parser/TransformParser.java

```diff
@@ -0,0 +1,342 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ververica.cdc.runtime.parser;
+
+import org.apache.flink.api.common.io.ParseException;
+
+import com.ververica.cdc.common.schema.Column;
+import com.ververica.cdc.common.types.DataTypes;
+import com.ververica.cdc.common.utils.StringUtils;
+import com.ververica.cdc.runtime.operators.transform.ColumnTransform;
+import com.ververica.cdc.runtime.parser.validate.FlinkCDCOperatorTable;
+import com.ververica.cdc.runtime.parser.validate.FlinkCDCSchemaFactory;
+import com.ververica.cdc.runtime.typeutils.DataTypeConverter;
+import org.apache.calcite.config.CalciteConnectionConfigImpl;
+import org.apache.calcite.config.Lex;
+import org.apache.calcite.jdbc.CalciteSchema;
+import org.apache.calcite.plan.RelOptCluster;
+import org.apache.calcite.plan.hep.HepPlanner;
+import org.apache.calcite.plan.hep.HepProgramBuilder;
+import org.apache.calcite.prepare.CalciteCatalogReader;
+import org.apache.calcite.rel.RelNode;
+import org.apache.calcite.rel.RelRoot;
+import org.apache.calcite.rel.type.RelDataType;
+import org.apache.calcite.rel.type.RelDataTypeField;
+import org.apache.calcite.rel.type.RelDataTypeSystem;
+import org.apache.calcite.rex.RexBuilder;
+import org.apache.calcite.sql.SqlBasicCall;
+import org.apache.calcite.sql.SqlIdentifier;
+import org.apache.calcite.sql.SqlKind;
+import org.apache.calcite.sql.SqlNode;
+import org.apache.calcite.sql.SqlSelect;
+import org.apache.calcite.sql.fun.SqlStdOperatorTable;
+import org.apache.calcite.sql.parser.SqlParseException;
+import org.apache.calcite.sql.parser.SqlParser;
+import org.apache.calcite.sql.type.SqlTypeFactoryImpl;
+import org.apache.calcite.sql.util.SqlOperatorTables;
+import org.apache.calcite.sql.validate.SqlConformanceEnum;
+import org.apache.calcite.sql.validate.SqlValidator;
+import org.apache.calcite.sql.validate.SqlValidatorUtil;
+import org.apache.calcite.sql2rel.SqlToRelConverter;
+import org.apache.calcite.sql2rel.StandardConvertletTable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Properties;
+import java.util.stream.Collectors;
+
+/** Use Flink's calcite parser to parse the statement of flink cdc pipeline transform. */
+public class TransformParser {
+    private static final Logger LOG = LoggerFactory.getLogger(TransformParser.class);
+    private static final String DEFAULT_SCHEMA = "default_schema";
+    private static final String DEFAULT_TABLE = "TB";
+    public static final String DEFAULT_DATABASE_NAME = "__database_name__";
+    public static final String DEFAULT_TABLE_NAME = "__table_name__";
```

Comment:
Have you considered our three level table path i.e `namespaceName.schemaName.tableName`?